godot-core/src/builtin/vectors/vector2i.rs

```diff
@@ -137,4 +168,25 @@ mod test {
 
         crate::builtin::test_utils::roundtrip(&vector, expected_json);
     }
+
+    #[test]
+    fn axis_min_max() {
+        let x_max = Vector2i::new(10, 5);
+        let y_max = Vector2i::new(5, 10);
+
+        let x_min = Vector2i::new(-5, 5);
+        let y_min = Vector2i::new(5, -5);
+
+        let max_eq = Vector2i::new(15, 15);
+        let min_eq = Vector2i::new(15, 15);
+
+        assert_eq!(x_max.max_axis(), Some(Vector2Axis::X));
+        assert_eq!(y_max.max_axis(), Some(Vector2Axis::Y));
+
+        assert_eq!(x_min.min_axis(), Some(Vector2Axis::X));
+        assert_eq!(y_min.min_axis(), Some(Vector2Axis::Y));
+
+        assert_eq!(max_eq.max_axis(), None);
+        assert_eq!(min_eq.min_axis(), None);
+    }
```

Comment:
Since you only use each variable once, it would be easier to inline them, as the condition would be self-contained. I.e. this test would just have 6 assertions and no other statements.